common/thorhelper/thorsoapcall.cpp

```diff
@@ -2250,9 +2260,19 @@ class CWSCAsyncFor : implements IWSCAsyncFor, public CInterface, public CAsyncFo
         if (checkContentDecoding(dbgheader, response, contentEncoding))
             decodeContent(contentEncoding.str(), response);
         if (soapTraceLevel > 6 || master->logXML)
-            master->logctx.mCTXLOG("%s: LEN=%d %sresponse(%s%s)", getWsCallTypeName(master->wscType),response.length(),chunked?"CHUNKED ":"", dbgheader.str(), response.str());
-        else if (soapTraceLevel > 8)
-            master->logctx.mCTXLOG("%s: LEN=%d %sresponse(%s)", getWsCallTypeName(master->wscType),response.length(),chunked?"CHUNKED ":"", response.str()); // not sure this is that useful but...
+        {
+            if ( (master->logXML) && (soapSepString.length() > 0) )
+            {
+                StringBuffer response2(dbgheader);
+                response2.append(response);
+                response2.replaceString("\r\n", soapSepString.str());
+                response2.replaceString("\r", soapSepString.str());
+                response2.replaceString("\n", soapSepString.str());
```

Comment:
Quite expensive (to build up new StringBuffer with dbgheader+response), then call multiple replaceString's which will walk the whole response for each call.

It may be worth adding a function to append to a StringBuffer replacing the sequences with the new separator (or removing them) as it goes.
An efficient implementation could be quite simple if it only removed (meaning the target was guaranteed to be smaller than source).

Reply:
NOTE this PR to optimize replaceString: https://github.com/hpcc-systems/HPCC-Platform/pull/19187